Tests/Source/ModelTester.cpp

```diff
@@ -12,95 +12,95 @@ using namespace std::chrono_literals;
+        m_data.push_back(sample);
       }
+      if(m_data.empty()) {
+        return;
+      }
+      compute_norm();
+      normalize_data();
+    }
 
-      double predict(Arguments arguments) const {
-        normalize_arguments(arguments);
-        struct Candidate {
-          double distance;
-          double result;
-        };
-        auto candidate1 = std::optional<Candidate>();
-        auto candidate2 = std::optional<Candidate>();
-        for(const auto& sample : m_data) {
-          auto distance = calculate_squared_distance(sample.m_arguments,
-            arguments);
-          if(!candidate1) {
-            candidate1 = { distance, sample.m_result };
-          } else if(distance < candidate1->distance) {
-            candidate2 = candidate1;
-            candidate1 = { distance, sample.m_result };
-          } else if(!candidate2 || distance < candidate2->distance) {
-            candidate2 = { distance, sample.m_result };
-          }
-        }
-        if(!candidate2) {
-          return candidate1->result;
-        } else {
-          return (candidate1->result + candidate2->result) / 2.;
+    double predict(Arguments arguments) const {
+      normalize_arguments(arguments);
+      struct Candidate {
+        double distance;
+        double result;
+      };
+      auto candidate1 = std::optional<Candidate>();
+      auto candidate2 = std::optional<Candidate>();
+      for(const auto& sample : m_data) {
+        auto distance = calculate_squared_distance(sample.m_arguments,
+          arguments);
+        if(!candidate1) {
+          candidate1 = { distance, sample.m_result };
+        } else if(distance < candidate1->distance) {
+          candidate2 = candidate1;
+          candidate1 = { distance, sample.m_result };
+        } else if(!candidate2 || distance < candidate2->distance) {
+          candidate2 = { distance, sample.m_result };
         }
       }
+      if(!candidate2) {
+        return candidate1->result;
+      } else {
+        return (candidate1->result + candidate2->result) / 2.;
+      }
+    }
 
-    private:
-      void compute_norm() {
-        m_min_values = m_data[0].m_arguments;
-        m_max_values = m_data[0].m_arguments;
-        for(auto i = std::size_t(0); i < m_data[0].m_arguments.size(); ++i) {
-          auto [min_it, max_it] = std::minmax_element(m_data.begin(),
-              m_data.end(), [&](const auto& lhs, const auto& rhs) {
-            return lhs.m_arguments[i] < rhs.m_arguments[i];
-          });
-          m_min_values[i] = min_it->m_arguments[i];
-          m_max_values[i] = max_it->m_arguments[i];
-        }
+  private:
+    void compute_norm() {
+      m_min_values = m_data[0].m_arguments;
+      m_max_values = m_data[0].m_arguments;
+      for(auto i = std::size_t(0); i < m_data[0].m_arguments.size(); ++i) {
+        auto[min_it, max_it] = std::minmax_element(m_data.begin(),
+          m_data.end(), [&](const auto& lhs, const auto& rhs) {
```

Comment:
Use a non-capturing lambda.

Reply:
It captures the index `i`.

Reply:
Correct.

Reply:
Do you mean I should capture `i` by copy instead of using reference all-capture? You mentioned limited captures should only be used when mixing capture types (copy vs reference).